onnxruntime/test/providers/internal_testing/internal_testing_partitioning_tests.cc

```diff
@@ -0,0 +1,325 @@
+        ++entry->second;
+      } else {
+        unsupported_ops[node.OpType()] = 1;
+      }
+    }
+  }
+
+  if (!unsupported_ops.empty()) {
+    bool add_comma = false;
+    for (const auto& pair : unsupported_ops) {
+      if (add_comma) {
+        oss << ",";
+      }
+
+      oss << pair.first << "(" << pair.second << ")";
+      add_comma = true;
+    }
+
+    unsupported_op_str = oss.str();
+  }
+
+  const auto& func_mgr = session->GetSessionState().GetFuncMgr();
+  NodeComputeInfo* compute_func = nullptr;
+
+  stats.num_nodes_not_handled = 0;
+  stats.num_compiled_nodes = 0;
+
+  // find the nodes downstream of the excluded nodes to check that they were assigned correctly
+  std::unordered_set<const Node*> excluded_nodes;
+  if (!stop_ops.empty()) {
+    for (const auto& node : graph.Nodes()) {
+      if (stop_ops.find(node.OpType()) != stop_ops.cend()) {
+        excluded_nodes.insert(&node);
+
+        // add all the downstream nodes to the excluded list
+        std::queue<const Node*> nodes_to_add;
+        nodes_to_add.push(&node);
+        while (!nodes_to_add.empty()) {
+          const Node* cur_node = nodes_to_add.front();
+          nodes_to_add.pop();
+
+          std::for_each(cur_node->OutputNodesBegin(), cur_node->OutputNodesEnd(),
+                        [&nodes_to_add, &excluded_nodes](const Node& output_node) {
+                          nodes_to_add.push(&output_node);
+                          excluded_nodes.insert(&output_node);
+                        });
+        }
+      }
+    }
+  }
+
+  for (const auto& node : graph.Nodes()) {
+    if (stop_ops.empty() || excluded_nodes.find(&node) == excluded_nodes.cend()) {
+      EXPECT_EQ(ops.count(node.OpType()), size_t(0))
+          << "Nodes with supported op types should have been replaced. Node with type "
+          << node.OpType() << " was not.";
+    } else {
+      EXPECT_NE(node.GetExecutionProviderType(), utils::kInternalTestingExecutionProvider)
+          << "Node is downstream from a 'stop at' node and should not have been taken. Node:"
+          << node.Name();
+    }
+
+    if (node.GetExecutionProviderType() == utils::kInternalTestingExecutionProvider) {
+      EXPECT_STATUS_OK(func_mgr.GetFuncs(node.Name(), compute_func));
+      EXPECT_NE(compute_func, nullptr);
+      ++stats.num_compiled_nodes;
+    } else {
+      ++stats.num_nodes_not_handled;
+    }
+  }
+
+  stats.num_nodes_handled = num_nodes - stats.num_nodes_not_handled;
+
+  auto pad_str = [](std::string const& str, size_t len = 10) {
+    return (str.size() < len) ? str + std::string(len - str.size(), ' ') : str;
+  };
+
+  std::cout << pad_str(test_name, 25)
+            << ": Compiled=" << stats.num_compiled_nodes
+            << " Handled=" << stats.num_nodes_handled
+            << " NotHandled=" << stats.num_nodes_not_handled
+            << " UnsupportedOps=" << unsupported_op_str
+            << "\n";
+}
+
+TEST(InternalTestingEP, TestNnapiPartitioningMlPerfModels) {
```

Comment:
this test could be disabled instead and then we can ensure the code still compiles